rpc-tests/rpc/zkVerify/submitProof/index.test.ts

```diff
@@ -0,0 +1,109 @@
+                        console.log(`Transaction included in block (elapsed time: ${(Date.now() - startTime) / 1000} seconds)`);
+                        handleEvents(events, (data: any[]) => {
+                            proof_leaf = data[0].toString();
+                            attestation_id = data[1].toString();
+                            console.log(`Proof Verified:\n  - Attestation Id: ${attestation_id}\n  - Proof Leaf: ${proof_leaf}`);
+                        });
+
+                        if (dispatchError) {
+                            console.error(`Invalid ${proofType} transaction:`, dispatchError.toString());
+                            reject(dispatchError);
+                        } else {
+                            resolve();
+                        }
+                    }
+
+                    if (status.isFinalized) {
+                        console.log(`Block containing ${proofType} proof transaction finalized (elapsed time: ${(Date.now() - startTime) / 1000} seconds)`);
+                    }
+                });
+            });
+
+            await waitForAttestationId(attestation_id);
+
+            console.log(`Waiting for NewAttestation event...`);
+            const eventData = await waitForNewAttestation(api, 360000, attestation_id, startTime);
+            const [attestationId, proofsAttestation] = eventData;
+            expect(Number.isInteger(attestationId)).toBeTruthy();
+            expect(proofsAttestation).toMatch(/^0x[a-fA-F0-9]{64}$/);
+        }, 300000);
+
+        test(`should reject invalid ${proofType} proof upon finalization`, async () => {
+            const keyring = new Keyring({ type: 'sr25519' });
+            const account = keyring.addFromUri(process.env.PRIVATE_KEY as string);
+
+            const submitProof = api.tx[pallet].submitProof(invalidProof);
+
+            const result = await new Promise<boolean>((resolve, reject) => {
+                submitProof.signAndSend(account, ({ status, dispatchError }: SubmittableResult) => {
+                    if (status.isInBlock) {
+                        console.log(`Transaction included at blockHash ${status.asInBlock.toString()}`);
+                    }
+                    if (status.isFinalized) {
+                        if (dispatchError) {
+                            console.error('Transaction failed as expected due to an error.');
+                            resolve(true);
+                        } else {
+                            console.error('Transaction finalized without error, test failed.');
+                            reject(new Error('Transaction was expected to fail but did not.'));
+                        }
+                    }
+                }).catch(error => {
+                    console.error(`Transaction submission failed with an error: ${error}`);
+                    reject(error);
+                });
+            });
+
+            expect(result).toBeTruthy();
+        }, 300000);
```

Comment:
Also here

Reply:
Thanks i'll reduce this to 200 i think i usually see around 170s or so